mock/cpp_mock_scenarios/src/cpp_scenario_node.cpp

```diff
@@ -112,30 +113,14 @@ void CppScenarioNode::spawnEgoEntity(
     return configuration;
   }());
   api_.requestAssignRoute("ego", goal_lanelet_poses);
+
   using namespace std::chrono_literals;
-  std::atomic<bool> initialized(false);
-  auto initialize_thread = std::thread([&]() {
-    while (!api_.asFieldOperatorApplication("ego").engaged()) {
-      api_.updateFrame();
-      std::this_thread::sleep_for(std::chrono::duration<double>(1.0 / 20.0));
-    }
-    initialized.store(true);
-  });
-  std::atomic<bool> engaged(false);
-  auto engage_thread = std::thread([&]() {
-    while (!api_.asFieldOperatorApplication("ego").engaged()) {
+  while (!api_.asFieldOperatorApplication("ego").engaged()) {
+    if (api_.asFieldOperatorApplication("ego").engageable())
       api_.asFieldOperatorApplication("ego").engage();
-      std::this_thread::sleep_for(1000ms);
-    }
-    engaged.store(true);
-  });
-  while (!api_.asFieldOperatorApplication("ego").engaged() &&
-         !(initialized.load() && engaged.load())) {
-    // RCLCPP_INFO_STREAM(get_logger(), "Waiting for Autoware initialization...");
+    api_.updateFrame();
     std::this_thread::sleep_for(std::chrono::duration<double>(1.0 / 20.0));
   }
-  initialize_thread.join();
-  engage_thread.join();
 }
```

Comment:
Here the idea of simplifying the "Waiting for Autoware initialization" state in CppScenarioNode is presented. Instead of creating two separate threads, one responsible for updating the simulation and the other responsible for engaging the concealer, I included both this actions in the current thread.

I do not see any reason why this part of code should not be simplified in this way, however if during the revision this simplification is found to be incorrect, please note it.